setup.py

```diff
@@ -28,21 +29,16 @@
 # SUPPORTED_ARCHS = NVIDIA_SUPPORTED_ARCHS.union(ROCM_SUPPORTED_ARCHS)
 
 
+def _is_cuda() -> bool:
+    return torch.version.cuda is not None
+
+
 def _is_hip() -> bool:
     return torch.version.hip is not None
 
 
 def _is_neuron() -> bool:
-    torch_neuronx_installed = True
-    try:
-        subprocess.run(["neuron-ls"], capture_output=True, check=True)
-    except FileNotFoundError:
-        torch_neuronx_installed = False
-    return torch_neuronx_installed
-
-
-def _is_cuda() -> bool:
-    return (torch.version.cuda is not None) and not _is_neuron()
+    return shutil.which("neuron-ls") is not None
```

Comment:
Just to let you know, I tried implementing just this line and it doesn't work on an AWS AMI as the neuron-ls package is installed, but if you're not on a neuron device then it still fails. What worked for me was to just change the errors caught on line 39 to include 'subprocess.CalledProcessError'.

Reply:
Thanks! updated my PR to reflect that